packages/flutter/lib/src/widgets/image.dart

```diff
@@ -1180,6 +1180,21 @@ class _ImageState extends State<Image> with WidgetsBindingObserver {
                 : null,
       ),
     );
+    if (widget.errorBuilder != null) {
```

Comment:
Does this support the following sequence?

1. widget build with `widget.errorBuilder != null`
2. widget rebuild with `widget.errorBuilder == null`
3. widget dispose
4. image failed

Reply:
That's a great point! I've added a test for it and adjusted code to satisfy this case.